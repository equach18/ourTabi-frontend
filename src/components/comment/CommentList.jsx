import CommentCard from "./CommentCard";

function CommentList({ comments, members }) {
  return (
    <div className="flex-1 overflow-y-auto space-y-4">
      {comments.length > 0 ? (
        comments.map((comment) => (
          <CommentCard
            key={comment.id}
            comment={comment}
          />
        ))
      ) : (
        <p className="text-gray-500">
          No comments yet. Start the conversation!
        </p>
      )}
    </div>
  );
}

export default CommentList;
