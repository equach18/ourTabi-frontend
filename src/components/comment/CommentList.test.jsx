import { render, screen } from "@testing-library/react";
import CommentList from "./CommentList";

// isolate CommentCard  testing
vi.mock("./CommentCard", () => ({
  default: ({ comment }) => <div>{comment.text}</div>,
}));

describe("CommentList", () => {
  const testComments = [
    { id: 1, text: "comment 1" },
    { id: 2, text: "comment 2" },
  ];

  it("matches snapshot", () => {
    const { asFragment } = render(<CommentList comments={testComments} />);
    expect(asFragment()).toMatchSnapshot();
  });
  
  it("renders a list of comments", () => {
    render(<CommentList comments={testComments} />);

    expect(screen.getByText("comment 1")).toBeInTheDocument();
    expect(screen.getByText("comment 2")).toBeInTheDocument();
  });

  it("renders fallback text when there are no comments to render", () => {
    render(<CommentList comments={[]} />);
    expect(
      screen.getByText(/no comments yet\. start the conversation/i)
    ).toBeInTheDocument();
  });
});
