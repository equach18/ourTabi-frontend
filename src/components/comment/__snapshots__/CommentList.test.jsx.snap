// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`CommentList > matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="flex-1 overflow-y-auto space-y-4"
  >
    <div>
      comment 1
    </div>
    <div>
      comment 2
    </div>
  </div>
</DocumentFragment>
`;
